src/Session/Session.php

```diff
@@ -101,9 +122,154 @@ public function setData($data)
      */
     public function getUser()
     {
-        $user_uuid = $this->get('user_uuid');
-        // TODO: Remove this direct instantiation to make this more testable
-        $user = new User((object)array('id' => $user_uuid));
-        return $user;
+        if ($this->loggedIn()) {
+            $user_id = $this->get('user_id');
+            // TODO: Remove this direct instantiation to make this more testable
+            $user = new User((object)array('id' => $user_id));
+            return $user;
+        }
+    }
+
+    /**
+     * Authorize the current user prior to running a command.  The
+     * Annotated Commands hook manager will call this function during
+     * the pre-validate phase of any command that has an 'authorize'
+     * annotation.
+     *
+     */
+    public function ensureLogin()
```

Comment:
Yeah. I figured there was a more elegant way to do this but since I was already stretching the responsibility of the `Session` class about as far as I was willing to stretch it I figured I'd just keep this adapter in place and basically punt on this part. Maybe by the time we build our second and third and fourth custom hook we will come up with an elegant way to register them.